### PR TITLE
Updates to README.md file

### PR DESCRIPTION
**Pull request by ljhwang**
Multiple updates to the file:

- removes version numbers from title and license so this does not have to be updated between versions.
- adds license tag
- introduces new section More information patterned after ASPECT. This has room to name maintainers.
- adds the "and newer" to the GPL3 license statement at the end.  

This should be reviewed by @feathern

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,10 +32,20 @@ The following documents form the Rayleigh documentation.
 | https://rayleigh-documentation.readthedocs.io/en/latest/ | A combined online documentation |
 | https://rayleigh-documentation.readthedocs.io/en/latest/doc/source/diagnostic_codes/qcodes.html | Online tables of Rayleigh output menu codes |
 
+More information
+----------------
+- For questions on the source code of Rayleigh, portability, installation, new or existing features, etc., use the [Rayleigh forum](https://community.geodynamics.org/c/rayliegh/5). This forum is where the Rayleigh users and developers all hang out.
```
Comment (gassmoeller) [inline README.md:37]:
```suggestion
- For questions on the source code of Rayleigh, portability, installation, new or existing features, etc., use the [Rayleigh forum](https://community.geodynamics.org/c/rayleigh/5). This forum is where the Rayleigh users and developers all hang out.
```

**Review by gassmoeller — COMMENTED** [review]
Looks good to me except for the one typo.

**Comment by ljhwang** [comment]
This should resolved #190

**Review by gassmoeller — APPROVED** [review]
Looks good. It seems the tester only lost connection to conda, the restarted test worked.

**Comment by gassmoeller** [comment]
Thanks!

**Pull request merged**